src/test/util_tests.cpp

```diff
@@ -121,7 +121,8 @@ BOOST_AUTO_TEST_CASE(util_ParseParameters)
     testArgs.ParseParameters(1, (char**)argv_test);
     BOOST_CHECK(testArgs.GetMapArgs().empty() && testArgs.GetMapMultiArgs().empty());
 
-    testArgs.ParseParameters(5, (char**)argv_test);
+    static_assert(7 * sizeof(argv_test[0]) == sizeof(argv_test), "argv_test length has changed");
```

Comment:
IMO remove this assert.

Reply:
The assert doesn't have any runtime cost, and documents/enforces the change as well as helping avoid the same mistake happening again, so seems better to have it to me?

An alternative approach is using templates to directly determine the size of the array:

    template<size_t N>
    inline void ParseParameters(int argc, const char* (&argv)[N])
    {
        assert(argc >= 0 && (size_t) argc < N);
        ArgsManager::ParseParameters(argc, (char**) argv);
    }
    template<size_t N>
    inline void ParseParameters(const char* (&argv)[N])
    {
        ArgsManager::ParseParameters(N, (char**) argv);
    }
    ...
    testArgs.ParseParameters(1, argv_test); // to test only some args
    testArgs.ParseParameters(argv_test); // to test all args

Reply:
Since passing an `argc` that does not specify the size of `argv` is only ever used in tests, you could pass `argv_test.size()` instead of `7` by switching to `const std::array<const char*, 7> argv_test = {... braced-init-list ...}`

Reply:
That makes it robust to future changes, but not robust to future changes that remove the trailing or the two trailing args. Imo such buggy changes must be covered by code review. Not excessive documentation. Note that we are talking about a function that spans about 20 lines... If code review does not work on that scale we have far worse issues.

Reply:
Removed